processor/src/main/java/org/mapstruct/ap/model/DecoratorConstructor.java

```diff
@@ -29,13 +28,14 @@
  *
  * @author Gunnar Morling
  */
-public class DecoratorConstructor extends ModelElement {
+public class DecoratorConstructor extends Field {
```

Comment:
It is.. However its confirming the current practice..(that's why I ran into problems)..  I'll change that.